Add fake start and end pages

CJWaszczuk opened the pull request:
<!--
  Is this related to a ticket in Jira? If so, add Smart Commit commands to the
  PR title to be automatically included in the merge commit if the PR is merged.

  For example:

  MT3-221 #ready-for-test #comment Create a pull request template

  https://confluence.atlassian.com/fisheye/using-smart-commits-960155400.html
  -->

# What?
Added start, submit and end pages.
<!--
  What new features or changes does this pull request contain?

  Include before and after screenshots, if appropriate.
  -->

# Why?
When beginning a process, a user will need to be online when they reach the index page, so that they can download any necessary data for when they go offline. They will also need to be online when they submit their data to their managers. And the end page is to indicate that they may now go on to complete another task. 

<!--
  What motivates these changes?

  Include any user stories driving this feature, if appropriate.
  -->

# Screenshots
## Start screen
### While loading data
![image](https://user-images.githubusercontent.com/8092247/70445155-03884b80-1a93-11ea-9595-1c1042c417a4.png)
### When data is loaded
![image](https://user-images.githubusercontent.com/8092247/70445135-f53a2f80-1a92-11ea-8a11-441d80eddc61.png)
## Submit screen
### When offline 

![image](https://user-images.githubusercontent.com/8092247/70446911-bd80b700-1a95-11ea-8813-5b1b33102a7f.png)

### When online 
![image](https://user-images.githubusercontent.com/8092247/70446871-ac37aa80-1a95-11ea-895f-77e649ed2cf0.png)

## End screen 
![image](https://user-images.githubusercontent.com/8092247/70446642-52cf7b80-1a95-11ea-876d-81c729c178f8.png)

CJWaszczuk committed:
Import and set up lbh-frontend-react

We add `babel-plugin-module-resolver` and TypeScript config to alias
the deep paths of `lbh-frontend-react` until that library exports in
a clean way.

CJWaszczuk commented:
Forgot to draft!

erbridge commented on __tests__/features/index-page.feature:
```diff
@@ -4,10 +4,15 @@ Feature: Index page
-    Then the page title should be "Tenancy & Household Check - Index"
-    And I should see "This will be a start page!" on the page
+    Then the page title should be "Tenancy & Household Check"
+    And I should see "Please wait until the 'Go' button" on the page
```
Didn't we change this to be a heading or something?

CJWaszczuk replied:
Do you mean the title for the remultiform pages? Or I may be misunderstanding

erbridge replied:
I mean to test for the presence of a heading, rather than some random body test that's likely to change.

CJWaszczuk replied:
Changed to the title of the page, which just so happens to be 'Tenancy & Household Check' also

erbridge commented on __tests__/layouts/MainLayout.test.tsx:
```diff
@@ -3,12 +3,22 @@ import { create } from "react-test-renderer";
   );
 
-  expect(component.toJSON()).toMatchSnapshot();
+  expect(component).toMatchSnapshot();
```
Let's switch to inline snapshots, in this project, too.

erbridge commented:
Only a quick review. Let me know what it's ready for a proper look, please.

erbridge commented on __tests__/features/end-page.feature:
```diff
@@ -0,0 +1,14 @@
+
+  As a user
+  I want to visit the end page
+  So I know my data is submitted
```
```suggestion
  So that I know it's safe to close it
```

erbridge commented on __tests__/features/end-page.feature:
```diff
@@ -0,0 +1,14 @@
+Feature: End page
+
+  As a user
+  I want to visit the end page
```
```suggestion
  I want to know that the process was submitted
```

erbridge commented on __tests__/features/index-page.feature:
```diff
@@ -4,10 +4,15 @@ Feature: Index page
   I want to visit the index page
```
```suggestion
   I want the index page to fetch all the data it will need
```

erbridge commented on __tests__/features/index-page.feature:
```diff
@@ -4,10 +4,15 @@ Feature: Index page
   I want to visit the index page
   So that I can start using the service
```
```suggestion
   So that I can work offline
```

erbridge commented on __tests__/features/submit-page.feature:
```diff
@@ -0,0 +1,25 @@
+Feature: Submit page
+
+  As a user
+  I want to visit the submit page
```
```suggestion
  I want the submit page to submit the completed process
```

erbridge commented on __tests__/features/submit-page.feature:
```diff
@@ -0,0 +1,25 @@
+
+  As a user
+  I want to visit the submit page
+  So that I can submit my data
```
```suggestion
  So that my work will be recorded
```

erbridge commented on layouts/MainLayout.tsx:
```diff
@@ -1,17 +1,29 @@
+        width: 60%;
+        margin: 0 auto;
+      }
+    `}</style>
```
Can we use `Main` now?

erbridge commented on layouts/MainLayout.tsx:
```diff
@@ -1,17 +1,29 @@
 
-type LayoutProps = {
-  children: ReactNode;
+type Props = {
```
```suggestion
interface Props {
```

Not a big deal. Only change if you're here anyway.

erbridge commented on package.json:
```diff
@@ -25,9 +25,12 @@
   "dependencies": {
     "@zeit/next-css": "^1.0.1",
+    "cross-fetch": "^3.0.4",
+    "lbh-frontend-react": "file:../lbh-frontend-react",
```
Oops.

erbridge commented on pages/_app.tsx:
```diff
@@ -1,5 +1,13 @@
+  components: {
+    Link: props => <Link {...props} />
+  }
+});
```
Remove this. It doesn't work.

erbridge reviewed (commented):
I think the pages need some really simple snapshot tests, too.

erbridge commented on layouts/MainLayout.tsx:
```diff
@@ -1,17 +1,29 @@
+      <link
+        href="https://fonts.googleapis.com/css?family=Montserrat:200,300,400,500,700&display=swap"
+        rel="stylesheet"
+      />
```
We have something to track that we need to fix this, right? Maybe add an issue to `lbh-frontend-react` or `lbh-frontend` (or both?).

erbridge committed:
Switch to using bound methods to alias WebDriver

This means we inherit the types from the wrapped driver, meaning we
don't need to painstakingly reimplement and update them.

erbridge committed:
Create start and end steps with fake offline syncing

I have left the offline tests commented out as we don't currently have a
good way to control the network of the running browser. We also need to
consider how this should work for server side rendering, though perhaps
if it came from the server, we should already know they're online?

CJWaszczuk commented on __tests__/features/prefetch-process-data.feature:
```diff
@@ -0,0 +1,26 @@
+Feature: Prefetch process data
+
+  As a user
+  I want to the service to prefetch the data needed for the process
```
'to the' - 'the'

erbridge committed:
Switch to requiring developers to install the webdrivers themselves

ChromeDriver's version needs to match the installed version of Chrome,
so it's better to require developers to maintain it themselves.
